File: backend/src/main/java/com/yabdioglu/ecommerce/dto/Purchase.java
package com.yabdioglu.ecommerce.dto;

import com.yabdioglu.ecommerce.entity.Address;
import com.yabdioglu.ecommerce.entity.Customer;
import com.yabdioglu.ecommerce.entity.Order;
import com.yabdioglu.ecommerce.entity.OrderItem;
import lombok.Data;

import java.util.Set;

@Data
public class Purchase {

    private Customer customer;
    private Address shippingAddress;
    private Address billingAddress;
    private Order order;
    private Set<OrderItem> orderItems;

}

File: angular-ecommerce/src/app/common/purchase.ts
import { Address } from "./address";
import { Customer } from "./customer";
import { Order } from "./order";
import { OrderItem } from "./order-item";

export class Purchase {
    customer: Customer;
    shippingAdress: Address;
    billingAddress: Address;
    order: Order;
    orderItems: OrderItem[];
}
